### PR TITLE
feat: file-templates as array in React render engine

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

- Render an array of File component from React render engine - file templates feature.
- Add documentation for new feature.
- Add unit tests for React.
- Increase timeout in tests to 2m :(

Docs part:

### React

The above way of rendering **file templates** works for both `nunjucks` and `react` render engine, but `react` also has another, more generic way to render multiple files. It is enough to return an array of `File` components in the rendering component. Check example:

```tsx
export default function({ asyncapi }) {
  return [
    <File name={`file1.html`}>Content</File>,
    <File name={`file2.html`}>Content</File>
  ]
}
```

**Related issue(s)**
Part of https://github.com/asyncapi/generator-react-sdk/issues/10

### CODE REVIEW

File: docs/authoring.md [hunk 1]
```diff
@@ -61,16 +63,44 @@ components:
...
+
 and
 ```
 Schema name is 'people' and properties are:
 - id
 ```
+
+### React
+
+The above way of rendering **file templates** works for both `nunjucks` and `react` render engine, but `react` also has another, more generic way to render multiple files. It is enough to return an array of `File` components in the rendering component. Check example:
```
Comment (jonaslagoni) [inline docs/authoring.md:81]:
```suggestion
The above way of rendering **file templates** works for both `nunjucks` and `react` render engines, but `react` also has another, more generic way to render multiple files. It is enough to return an array of `File` components in the rendering component. See the following example:
```

File: docs/authoring.md [hunk 2]
```diff
@@ -61,16 +63,44 @@ components:
...
+        <SchemaFile schemaName={schemaName} schema={schema} />
+      </File>
+    );
+  });
+}
+
+function SchemaFile({ schemaName, schema }) {
+  // implementation...
+}
+```
```
Comment (jonaslagoni) [inline docs/authoring.md:102]: I think this is too large of an example, what about: 

```suggestion
```tsx
export default function({ asyncapi }) {
  return [
    <File name={`file1.html`}>//Content</File>,
    <File name={`file2.html`}>//Content</File>
  ]
}
```
```

**Review by jonaslagoni — CHANGES_REQUESTED** [review]
Just some small comments. In hindsight, I should have added tests for the renderer when first integrating the feature. Gonna create a follow up issue about that.

**Comment by jonaslagoni** [comment]
@magicmatatjahu you have a problem with SonarCloud 😄

**Comment by derberg** [comment]
@magicmatatjahu will you also do a follow-up in the template for templates, to use this new way for file templates?

**Comment by magicmatatjahu** [comment]
@derberg Of course, I will, but I wanna wait for merge your PR to write tests for an array of Files - your PR has architecture for tests.

**Comment by jonaslagoni** [comment]
@magicmatatjahu you have a little problem with checks 😄

**Comment by derberg** [comment]
@magicmatatjahu you need to increase timeouts in the test project tests to a larger number, these tests are really long-running :(

**Review by jonaslagoni — DISMISSED** [review]
👍

**Review by jonaslagoni — APPROVED** [review]

**Review by jonaslagoni — APPROVED** [review]

**Pull request merged**